csrc/multidevice/utils.cpp

```diff
@@ -116,27 +116,19 @@ std::unordered_map<IterDomain*, int64_t> mapIterDomainToTensorAxis(
   return id_to_axis;
 }
 
-} // namespace
-
-int64_t getShardedLogicalAxis(
-    const TensorView* tv,
-    const ParallelType parallel_type) {
```

Comment:
We should also try to decide which domain according to parallel_type, so we have fewer knobs. My gut feeling is that "DIDs => loop and Stream => allocation" works universally. I failed to see why it's necessary to special-case unshardedSizes.

Reply:
Yes, from my observations so far.

Reply:
For logical domain sharding with stream parallelization, we need the `getAllocationDomain` instead of `getMaybeAllocationDomain`, though

Reply:
> For logical domain sharding with stream parallelization, we need the getAllocationDomain instead of getMaybeAllocationDomain, though

I failed to understand this statement. I suspect you were trying to say that, for logical domain sharding, we can't even represent a TensorView with stream-parallel loop and non-stream-parallel allocation. But I failed to see how it's connected to getAllocationDomain vs getMaybeAllocationDomain. 

Also, be aware of the difference between "MultiDeviceExecutor chose to implement foo" and "we need foo".